setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -16,11 +17,26 @@ class ClassesScanner implements ClassesScannerInterface
     protected $excludePatterns = [];
 
     /**
+     * @var array
+     */
+    private $fileResults = [];
+
+    /**
+     * @var string
+     */
+    private $generationDirectory;
+
+    /**
      * @param array $excludePatterns
+     * @param string $generationDirectory
      */
-    public function __construct(array $excludePatterns = [])
+    public function __construct(array $excludePatterns = [], DirectoryList $directoryList = null)
     {
         $this->excludePatterns = $excludePatterns;
+        if ($directoryList === null) {
+            $directoryList = ObjectManager::getInstance()->get(DirectoryList::class);
+        }
+        $this->generationDirectory = $directoryList->getPath(DirectoryList::GENERATION);
```

Comment:
it's not allowed to make function calls in the constructor, just assignments to class variables.

here we receive  $directoryList and after that make a call to get Path